src/Renderer/Shader/Chunk/projective_texturing_pars_fragment.glsl

```diff
@@ -0,0 +1,62 @@
+uniform sampler2D projectiveTexture[NUM_TEXTURES];
+varying vec4      projectiveTextureCoords[NUM_TEXTURES];
+uniform float     projectiveTextureAlphaBorder;
+
+struct Distortion {
+    vec2 size;
+#if USE_DISTORTION
+    vec2 pps;
+    vec4 polynom;
+    vec3 l1l2;
+#endif
+};
+
+uniform Distortion projectiveTextureDistortion[NUM_TEXTURES];
+
+float getAlphaBorder(vec2 p)
+{
+    vec2 d = clamp(projectiveTextureAlphaBorder*min(p,1.-p), 0., 1.);
+    return min(d.x,d.y);
+}
+
+#if USE_DISTORTION
+void distort(inout vec2 p, vec4 polynom, vec2 pps)
+{
+    vec2 v = p - pps;
+    float v2 = dot(v,v);
+    if(v2>polynom.w) p = vec2(-1.);
```

Comment:
Some spaces in shaders too

Reply:
ok, I've added space in shaders to for more readability :)